source-academy/frontend
# Categorise assessment achievements

### Description

The number of assessment achievements was getting large, making the achievements dashboard rather cluttered up. 

The change causes completed assessment achievements to be group into one dropdown achievement group, which greatly reduces the number of achievements listed in the dashboard. 

Uncompleted assessments do not get included in the dropdown, and are listed right at the top, sorted by how close the deadline is, to serve as a reminder to users that the assessments might be expiring soon. 

The behaviour of prerequisites in achievements has been changed. Prerequisites now no longer appear outside of their dropdown. This was done so that the assessment achievements only appeared in its dropdown. The parent achievement's view now also no longer displays the goals of its prerequisites. Those can still be found when selecting the prerequisite from the dropdown.

To prevent any odd errors from popping up due to this change, the adding of prerequisites to an achievement has been temporarily disabled. This should pose few problems as the prerequisite feature is frankly underdeveloped and not a very useful feature. The prerequisites can be reenabled when the feature is more fleshed out, or perhaps removed in the future.

### Type of change


### How to test

Upload a few assessments of different types, submit and grade some of them, and ensure that the completed assessment achievements appear in a dropdown called 'Completed Missions', etc. 

Ensure that uncompleted assessment achievements are listed on top of the dropdowns, and all other achievements are listed below the dropdowns. Also ensure that the uncompleted assessments are listed by deadline, with the closest deadline on top.

Also ensure that the XP of these achievements are counted properly in the progress bar at the top left of the achievement dashboard. 

### Checklist
